src/fdc3/dotnet/DesktopAgent.Client/src/MorganStanley.ComposeUI.Fdc3.DesktopAgent.Client/Infrastructure/DesktopAgentClient.cs

```diff
@@ -27,35 +29,94 @@ namespace MorganStanley.ComposeUI.Fdc3.DesktopAgent.Client.Infrastructure;
 internal class DesktopAgentClient : IDesktopAgent
 {
     private readonly IMessaging _messaging;
+    private readonly ILoggerFactory _loggerFactory;
     private readonly ILogger<DesktopAgentClient> _logger;
     private readonly string _appId;
     private readonly string _instanceId;
-    private readonly JsonSerializerOptions _jsonSerializerOptions = SerializerOptionsHelper.JsonSerializerOptions;
+    private readonly IChannelFactory _channelFactory;
+    private readonly IMetadataClient _metadataClient;
+    private readonly ConcurrentDictionary<IListener, Func<string, ChannelType, CancellationToken, ValueTask>> _contextListenersWithSubscriptionLastContextHandlingActions = new();
+
+    private IChannel? _currentChannel;
+    private readonly SemaphoreSlim _currentChannelLock = new(1, 1);
+
+    private readonly ConcurrentDictionary<string, IChannel> _userChannels = new();
```

Comment:
UserChannels are never loaded.
It is always accessed inside a lock, no need to make it concurrent.